### PR TITLE
Use svgr for SVGs

### PR DESCRIPTION
**Pull request by swissspidy**
Less magic, less dependencies, less security warnings.

See https://github.com/WordPress/gutenberg/pull/18243

### CODE REVIEW

**Comment by barklund** [comment]
What's the improvement here? Don't really see it in the linked GB issue.

**Comment by swissspidy** [comment]
We currently use s combination of a webpack loader and a Babel plugin for this, which are both pretty much unmaintained and also cause warnings to be raised by npm‘s audit command.

The linked PR shows that we can simplify this by using the more maintained svgr library which is now integrated in the wp-scripts suite of tools.

**Comment by barklund** [comment]
> We currently use s combination of a webpack loader and a Babel plugin for this, which are both pretty much unmaintained and also cause warnings to be raised by npm‘s audit command.
> 
> The linked PR shows that we can simplify this by using the more maintained svgr library which is now integrated in the wp-scripts suite of tools.

Sounds good with me!

The syntax is a little crude though - is there anyway to make the named `ReactComponent` export the default one? Though [it doesn't sound like it](https://create-react-app.dev/docs/adding-images-fonts-and-files/#adding-svgs).

**Comment by swissspidy** [comment]
I thought not, but maybe  it is possible: https://react-svgr.com/docs/webpack/#usage

I'll check again.

**Edit**: looks like that's indeed intended behavior: the default export is for 
the source link that can be used in an `img`, and `ReactComponent` is for using as a component

**Comment by swissspidy** [comment]
Just noticed some issues where icons are cropped. I'll take a look at that.

**Edit:** resolved in 2543566a9a50d3446a49ae39760ff51555251d32

**Comment by dvoytenko** [comment]
@swissspidy How did you get these SVG changes done? Is there some linter/commit hook for that?

**Comment by swissspidy** [comment]
You mean the removal of width/height attributes? Manually. But there might be a svgr config for that. Will check!

And changing the imports, well, without that the editor wouldn‘t run. But perhaps the linter might help there too, will check that.

**Comment by dvoytenko** [comment]
> You mean the removal of width/height attributes? Manually. But there might be a svgr config for that. Will check!
> 
> And changing the imports, well, without that the editor wouldn‘t run. But perhaps the linter might help there too, will check that.

Width/height and currentColor. I think it'd be enough to simply error out to remove manually.

**Comment by swissspidy** [comment]
Sounds good! I‘ll see whether it warrants a new PR or not.

**Review by dvoytenko — APPROVED** [review]

**Comment by dvoytenko** [comment]
otherwise lgtm

**Review by barklund — APPROVED** [review]
If it's better, it's better!

#welcometotautologyclub

**Comment by swissspidy** [comment]
@dvoytenko Done in 76aff4473292c0abed15bc09073fa354e8d15d11 🎉

File: .svgo.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+plugins:
+  - removeViewBox: false
+  - removeDimensions: true
+  - convertColors:
+      currentColor: true
```
Comment (barklund) [inline .svgo.yml:5]: This is awesome magic! Love it!

**Pull request merged**